File: Frontend-React/src/services/cart.service.ts
import http from '../api/http';

const json = { headers: { 'Content-Type': 'application/json' } };

export const CartService = {
  getCart: () => http.get('/cart/get'),
  addItem: (data: { productId: any; quantity: any }) => http.post('/cart/add', data, json),
  updateItem: (itemId: any, data: { quantity: any }) => http.put('/cart/item/' + itemId, data, json),
  removeItem: (itemId: any) => http.delete('/cart/item/' + itemId),
  clearCart: () => http.delete('/cart/clear'),
  applyCoupon: (code: string) => http.post('/cart/coupon/apply', { code }, json),
  removeCoupon: () => http.delete('/cart/coupon/remove'),
  checkout: (data: { addressId: any; paymentMethod: string; deliveryInstructions?: string; useLoyaltyPoints?: boolean }) =>
    http.post('/cart/checkout', data, json),
};

File: BackEnd/com.inn.cafe/src/main/java/com/inn/cafe/service/CartService.java
package com.inn.cafe.service;

import com.inn.cafe.dto.AddToCartRequest;
import com.inn.cafe.dto.ApplyCouponRequest;
import com.inn.cafe.dto.CheckoutRequest;
import com.inn.cafe.dto.UpdateCartItemRequest;
import com.inn.cafe.wrapper.CartWrapper;
import org.springframework.http.ResponseEntity;

public interface CartService {
    ResponseEntity<CartWrapper> getCart();

    ResponseEntity<CartWrapper> addItem(AddToCartRequest request);

    ResponseEntity<CartWrapper> updateItem(Integer itemId, UpdateCartItemRequest request);

    ResponseEntity<CartWrapper> removeItem(Integer itemId);

    ResponseEntity<CartWrapper> clearCart();

    ResponseEntity<CartWrapper> applyCoupon(ApplyCouponRequest request);

    ResponseEntity<CartWrapper> removeCoupon();

    ResponseEntity<String> checkout(CheckoutRequest request);
}
